src/world/MapLoader.cpp

```diff
@@ -136,6 +124,30 @@ void MapLoader::InitNodes(const glArchivItem_Map& map, Exploration exploration)
             resource = Resource(Resource::Granite, mapResource - 0x58);
         else if(mapResource > 0x80 && mapResource < 0x90) // fish
             resource = Resource(Resource::Fish, 4);       // Use 4 fish
+        else
```

Comment:
What about simply `else if (mapResource == 0x20 || mapResource == 0x21 || waterEverywhere)` and then the old code?
Then we have no duplication and there can be different amounts of water (why not?)

Reply:
Thinking about it: Maybe don't change the MapLoader as this is the loader from the S2 maps. We might change this or add another map format.
Maybe add a method similar to `ConvertMineResourceTypes` to `GameWorldGame` which does the whole water adjustment (including the terrain stuff). That would be more flexible, what do you think?

Reply:
Code is moved to GameWorldGame.